### PR TITLE
global config: fix etcd client not found

### LINKED ISSUE #6860: panic: WatchGlobalConfig
### What did you do?
restart pd

### What did you expect to see?
no panic

### What did you see instead?
pd panic

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x20491a4]

goroutine 546 [running]:
github.com/tikv/pd/server.(*GrpcServer).WatchGlobalConfig(0x4002820d50, 0x4002f106c0, {0x3929bb0, 0x4002f0ab60})
        /mnt/data1/jenkins/workspace/build-common@4/go/src/github.com/pingcap/pd/server/grpc_service.go:2366 +0x144
github.com/pingcap/kvproto/pkg/pdpb._PD_WatchGlobalConfig_Handler({0x2b851e0?, 0x4002820d50}, {0x3926ee8, 0x4002ac9cc8})
        /root/go/pkg/mod/github.com/pingcap/kvproto@v0.0.0-20230727073445-53e1f8730c30/pkg/pdpb/pdpb.pb.go:10273 +0xd0
github.com/grpc-ecosystem/go-grpc-middleware.ChainStreamServer.func1.1({0x2b851e0?, 0x4002820d50?}, {0x3926ee8?, 0x4002ac9cc8?})
        /root/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:71 +0x9c
github.com/grpc-ecosystem/go-grpc-prometheus.(*ServerMetrics).StreamServerInterceptor.func1({0x2b851e0, 0x4002820d50}, {0x3927a28?, 0x40013d2cc0}, 0x20?, 0x4002f128c0)
        /root/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-prometheus@v1.2.0/server_metrics.go:121 +0x108
github.com/grpc-ecosystem/go-grpc-middleware.ChainStreamServer.func1.1({0x2b851e0?, 0x4002820d50?}, {0x3927a28?, 0x40013d2cc0?})
        /root/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:74 +0x74
go.etcd.io/etcd/etcdserver/api/v3rpc.newStreamInterceptor.func1({0x2b851e0, 0x4002820d50}, {0x3927a28, 0x40013d2cc0}, 0x4002ac9c80?, 0x4002f128c0)
        /root/go/pkg/mod/go.etcd.io/etcd@v0.5.0-alpha.5.0.20220915004622-85b640cee793/etcdserver/api/v3rpc/interceptor.go:237 +0x420
github.com/grpc-ecosystem/go-grpc-middleware.ChainStreamServer.func1({0x2b851e0, 0x4002820d50}, {0x3927a28, 0x40013d2cc0}, 0x4002ac9c80, 0x36851c0)
        /root/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.0.1-0.20190118093823-f849b5445de4/chain.go:79 +0x17c
google.golang.org/grpc.(*Server).processStreamingRPC(0x4002822480, {0x392df80, 0x4002823080}, 0x4002f08700, 0x400067cf60, 0x49a5860, 0x0)
        /root/go/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:1244 +0xb9c
google.golang.org/grpc.(*Server).handleStream(0x4002822480, {0x392df80, 0x4002823080}, 0x4002f08700, 0x0)
        /root/go/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:1317 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        /root/go/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:722 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /root/go/pkg/mod/google.golang.org/grpc@v1.26.0/server.go:720 +0xdc
```
### What version of PD are you using (`pd-server -V`)?
```
[root@tiup-0 v7.3.0-alpha-nightly-20230730]# ./pd-server -V
Release Version: v7.3.0-alpha
Edition: Community
Git Commit Hash: fe52361cf48a7f5ed8c69bcd02db77e25162f207
Git Branch: heads/refs/tags/v7.3.0-alpha
UTC Build Time:  2023-07-28 02:25:10
```

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Close #6860

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by bufferflies — APPROVED** [review]

File: server/server.go [hunk 1]
```diff
@@ -1940,3 +1940,8 @@ func (s *Server) GetTSOUpdatePhysicalInterval() time.Duration {
 func (s *Server) GetMaxResetTSGap() time.Duration {
 	return s.persistOptions.GetMaxResetTSGap()
 }
+
+// SetClient [JUST FOR TEST] sets the etcd client.
```
Comment (lhy1024) [inline server/server.go:1944]:
```suggestion
// SetClient sets the etcd client.
// Notes: it is only used test.
```

File: server/grpc_service.go [hunk 2]
```diff
@@ -2316,6 +2319,9 @@ func (s *GrpcServer) StoreGlobalConfig(_ context.Context, request *pdpb.StoreGlo
 // - `Names` iteratively get value from `ConfigPath/Name` but not care about revision
 // - `ConfigPath` if `Names` is nil can get all values and revision of current path
 func (s *GrpcServer) LoadGlobalConfig(ctx context.Context, request *pdpb.LoadGlobalConfigRequest) (*pdpb.LoadGlobalConfigResponse, error) {
+	if s.client == nil {
```
Comment (lhy1024) [inline server/grpc_service.go:2322]: does it only happen when the server is starting and no start etcd server?

Comment (HuSharp) [reply server/grpc_service.go:2323]: As far as I know, this should be the only case

**Review by lhy1024 — APPROVED** [review]

File: server/grpc_service.go (hunk 2 above)
Comment (rleungx) [reply server/grpc_service.go:2323]: Why does this case happen?

Comment (HuSharp) [reply server/grpc_service.go:2323]: When grpc comes in before etcd has finished booting.

**Pull request merged**